plugins/inputs/hugepages/hugepages_notlinux.go

```diff
@@ -1,3 +1,30 @@
 //go:build !linux
 
 package hugepages
+
+import (
+	_ "embed"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Hugepages struct {
+	Log telegraf.Logger `toml:"-"`
+}
+
+func (h *Hugepages) Init() error {
+	h.Log.Warn("current platform is not supported")
+	return nil
+}
+func (h *Hugepages) SampleConfig() string                { return sampleConfig }
+func (h *Hugepages) Gather(_ telegraf.Accumulator) error { return nil }
```

Comment:
I think this should be
```suggestion
func (*Hugepages) SampleConfig() string                { return sampleConfig }
func (*Hugepages) Gather(_ telegraf.Accumulator) error { return nil }
```

Same for all other plugins. What do you think?